app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/triage/MultipleDisposalsTriageController.scala

```diff
@@ -927,24 +928,72 @@ class MultipleDisposalsTriageController @Inject() (
-                    routes.MultipleDisposalsTriageController
-                      .disposalDateOfSharesSubmit(),
-                    state.fold(_ => false, _._1.isAmendReturn)
+                    routes.MultipleDisposalsTriageController.disposalDateOfSharesSubmit(),
+                    isAmendReturn(state)
                   )
                 )
               },
-              shareDisposalDate =>
-                if (
-                  answers
-                    .fold(_.completionDate, c => Some(c.completionDate))
-                    .contains(CompletionDate(shareDisposalDate.value))
-                )
-                  Redirect(
-                    routes.MultipleDisposalsTriageController.checkYourAnswers()
-                  )
-                else {
-                  val result =
-                    for {
+              { shareDisposalDate =>
+                val existingDisposalDate = answers.fold(_.completionDate, c => Some(c.completionDate))
+
+                existingDisposalDate match {
+                  case Some(existingDate) if existingDate.value === shareDisposalDate.value =>
+                    Redirect(routes.MultipleDisposalsTriageController.checkYourAnswers())
+
+                  case Some(existingDate) if isAmendReturn(state)                           =>
+                    val result = for {
+                      existingTaxYear <- taxYearService.taxYear(existingDate.value)
+                      updatedAnswers  <- EitherT.fromEither[Future](
+                                           Right(
+                                             answers
+                                               .unset(_.completionDate)
+                                               .copy(
+                                                 taxYear = existingTaxYear,
+                                                 completionDate = Some(CompletionDate(shareDisposalDate.value)),
+                                                 taxYearAfter6April2020 = Some(existingTaxYear.isDefined)
+                                               )
+                                           )
+                                         )
+                      newState         = updateState(
+                                           state,
+                                           updatedAnswers,
+                                           d =>
+                                             d.bimap(
+                                               multipleIndirect =>
+                                                 multipleIndirect.copy(
+                                                   exampleCompanyDetailsAnswers = multipleIndirect.exampleCompanyDetailsAnswers,
+                                                   yearToDateLiabilityAnswers = None,
+                                                   gainOrLossAfterReliefs = None
+                                                 ),
+                                               multiple =>
```

Comment:
you shouldn't be in a non-indirect disposal case in this endpoint since we're talking about the disposal date of shares